Fix auto-created BQ dataset path to include project ID.

mrkumar9001 opened the pull request:
What: Fully qualify the BQ path to include project ID.
Why: Fix usage for when target project is different than the default in the environment (set by gcloud/GCE).

Thanks for pointing out the tests. Fixed the two tests that were affected & fixed another set of tests that were failing.

mrkumar9001 committed:
Fix auto-created BQ dataset path to include project ID.

pblesi commented:
Can you please add a "What" and a "Why" in the description to explain the change?

Does it make sense to have any tests supporting this change?

mrkumar9001 committed:
Fix tests: add project to BQ dataset, fix nanos for mock obj.

* With last commit, BQ path needs to be fully qualified with project name.
* Tests were failing with error: "Mock object has no attribute 'nanos'".
  This is due to instance attributes set in __init__().  See
  https://stackoverflow.com/q/31709792.

pblesi commented:
LGTM.
